esp/logging/logginglib/loggingagentbase.cpp

```diff
@@ -154,174 +154,172 @@ void CLogContentFilter::filterLogContentTree(StringArray& filters, IPropertyTree
 
 IEspUpdateLogRequestWrap* CLogContentFilter::filterLogContent(IEspUpdateLogRequestWrap* req)
 {
-    Owned<IPropertyTree> scriptValues = req->getScriptValuesTree();
-    const char* logContent = req->getUpdateLogRequest();
+    bool noFilter = logBackEndReq && logBackEndResp && (groupFilters.length() < 1);
     Owned<IPropertyTree> logRequestTree = req->getLogRequestTree();
-    Owned<IPropertyTree> updateLogRequestTree = createPTree("UpdateLogRequest");
+    const char* logContent = req->getUpdateLogRequest();
+    if (logRequestTree || !isEmptyString(logContent))
+    {
+        if (noFilter)
+            return LINK(req);
+
+        if (!logRequestTree)
+            logRequestTree.setown(createPTreeFromXMLString(logContent));
+
+        Owned<IPropertyTree> filteredLogRequestTree = createPTree(logRequestTree->queryName());
+        bool logContentEmpty = true;
+        filterLogContentTree(groupFilters.item(0).getFilters(), logRequestTree, filteredLogRequestTree, logContentEmpty);
+        if (logContentEmpty)
+            throw MakeStringException(EspLoggingErrors::UpdateLogFailed, "The filtered content is empty.");
 
+        if (getEspLogLevel()>=LogMax)
+        {
+            StringBuffer updateLogRequestXML;
+            toXML(filteredLogRequestTree, updateLogRequestXML);
+            ESPLOG(LogMax, "filtered content and option: <%s>", updateLogRequestXML.str());
+        }
+        req->setLogRequestTree(filteredLogRequestTree.getClear());
```

Comment:
This could lead to confusion. If the input contained non-empty logContent, that text remains in the input and the input now contains a tree that should be assumed to not match (we expect the filters to filter). Either the two values should be equivalent or only one value should be be present on exit.

Previously, the value returned was always the string in a new instance. If changed to use a tree (when a tree was given on input), must ensure that the consumers of this data are prepared for a tree and not assuming the string will be populated.